app/javascript/Domain/AuctionCoordinator.js

```diff
@@ -0,0 +1,67 @@
+import Action from './action';
+
+import AffordableBonds from './UseCases/Nations/Bonds/AffordableBonds';
+
+export default class AuctionCoordinator {
+  #gameCoordinator;
+
+  #nationToAuction;
+  #participatingPlayersInOrder;
+
+  #auctionPlayerIndex = -1;
+
+  #isComplete = false;
+
+  constructor(gameCoordinator, nationToAuction, participatingPlayersInOrder) {
```

Comment:
Instead of performing all the auctions at once, each AuctionCoordinator instance will represent a single round of auction for 1 country only.

This flexibility should allow `IGC` to reuse the same code for even individual bond purchases + swiss bankers